File: signal-singapore-23/inbound.py
## Using Python to create a Flask server on localhost:8080 that exposes a single endpoint /inbound
## Use ngrok to expose this server to the internet


from flask import Flask, request
from dotenv import load_dotenv
from twilio.rest import Client
from twilio.twiml.messaging_response import MessagingResponse
import ngrok
import os

load_dotenv()

app = Flask(__name__)
port = 8080

account_sid = os.getenv('TWILIO_ACCOUNT_SID')
auth_token = os.getenv('TWILIO_AUTH_TOKEN')

client = Client(account_sid, auth_token)


@app.route("/inbound", methods=['GET', 'POST'])
def singapore():
    twiml = MessagingResponse()

    from_number = request.form.get("From", "").replace("whatsapp:", "")
    lookup = client.lookups.v2.phone_numbers(from_number).fetch(fields="sms_pumping_risk")

    number_score = lookup.sms_pumping_risk.sms_pumping_risk_score
    twiml.message(f"Ahoy, your personal score is {number_score}")

    return str(twiml)


if __name__ == "__main__":
    try:
        ngrok_tunnel = ngrok.connect(port, authtoken_from_env=True, domain="ngrok.anthonydellavecchia.com")
        print("Running live at: ", ngrok_tunnel.url())
    except Exception as e:
        print("Error while starting ngrok:", e)

    app.run(host='localhost', port=port)

File: archive/monday/inbound.js
// this file is an express server on 3000 that exposes a single endpoint /inbound
// it accepts POST requests with a urlencoded body
// it returns a 200 response with a JSON body
// it uses ngrok to expose the endpoint to the internet
// it has a second endpoint /log that logs the request body to the console

const express = require('express');
const bodyParser = require('body-parser');
const ngrok = require('ngrok');
const twilio = require('twilio');
const { default: TwiML } = require('twilio/lib/twiml/TwiML');
const app = express();

const client = twilio(
    process.env.TWILIO_ACCOUNT_SID,
    process.env.TWILIO_AUTH_TOKEN
);

app.use(bodyParser.urlencoded({ extended: false }));
app.use(bodyParser.json());

app.post('/inbound', async (req, res) => {
    console.log(req.body);

    const lookup = await client.lookups.v2
        .phoneNumbers(req.body.From.replace("whatsapp:", ""))
        .fetch({ fields: ['sms_pumping_risk'] });

    const twiml = new twilio.twiml.MessagingResponse();

    const numberScore = lookup.smsPumpingRisk.sms_pumping_risk_score;
    twiml.message(`You score is ${numberScore}!`);

    res.writeHead(200, { 'Content-Type': 'text/xml' });
    res.end(twiml.toString());
});

app.post('/log', (req, res) => {
    console.log(req.body);
    res.json({ success: true });
})

const port = 3000

app.listen(port, () => {
    console.log('Server is running on port 3000');
    ngrok.connect({ port, subdomain: "mobert" }).then(ngrokUrl => {
        console.log('Twilio webhook is live at', ngrokUrl);
    }).catch(error => {
        console.error('Error while starting ngrok', error);
    });
});